share/spice/npm/npm.js

```diff
@@ -13,17 +16,53 @@
         // if query still has both items from handle remainder, remove $clean version
         if (query.indexOf('/' !== -1)) {
             query = query.split('/')[0];
-        }       
+        }
+        
+        // if the first package matches the exact query, then we'll return a text view
+        // For broader queries where there isn't a match, we will return the tile view
+        if (api_result.results[0].package.name === query) {
+            packageByNameExists = true;
+        };
         
     DDG.require('moment.js', function(moment) {
+        
+            var tile_template = {
+                group: 'text',
+                detail: false,
+                item_detail: false,
+                options: {
+                    footer: Spice.npm.footer
+                },
+                variants: {
+                    tile: 'basic4'
+                }
+            }
+            
+            var text_template =  {
+                group: 'text',
+                options: {
+                    content: Spice.npm.content,
+                    moreAt: true,
+                    moreText: {
+                        href: 'https://www.npmjs.com/package/' + query
+                    }
+                }
+            }
+            
+            // Chose the template based on template match
+            var template = packageByNameExists ? text_template : tile_template;
+            var sourceURL = packageByNameExists ? 
+                                'https://www.npmjs.com/package/' + query : 
+                                'https://www.npmjs.com/search?q=' + query;
+
             Spice.add({
                 id: "npm",
                 name: "Software",
-                data: api_result.results,
+                data: packageByNameExists ? api_result.results[0] : api_result.results,
                 meta: {
                     sourceName: "npmjs",
                     itemType: (api_result.total === 1) ? 'npm package' : 'npm packages',
-                    sourceUrl: 'https://www.npmjs.com/search?q=' + query
+                    sourceUrl: sourceURL
                 },
                 normalize: function(item) {
                     
```

Comment:
@moollaza ok, I _think_ I may have a solution. So in the api request (https://api.npms.io/v2/search?q=math) there is a score hash:

```
score: {
   final: 0.11189642866204279,
   detail: {
      quality: 0.10634079778423813,
      popularity: 0.007711438366202446,
      maintenance: 0.2208433882817157
   }
}
```
If it's below say `4`, we could chuck it. What do you think?